src/Service/ServiceManagerConfig.php

```diff
@@ -150,6 +150,7 @@ public function __construct(array $config = [])
                 // implementations; they're dealt with later.
                 if ($instance instanceof ServiceLocatorAwareInterface
                     && ! $instance instanceof AbstractPluginManager
+                    && ! $instance->getServiceLocator()
```

Comment:
@Ocramius I think yes old versions always set servicelocator. 
```php
'ServiceLocatorAwareInitializer' => function ($instance, ServiceLocatorInterface $serviceLocator) {
    if ($instance instanceof ServiceLocatorAwareInterface) {
        $instance->setServiceLocator($serviceLocator);
    }
},
```
But what it change? Do you think this would mean BC break? If someone has BC issues with this would be really bad code. Also AbstractPluginManager then have BC break it does not set service locator if already set in current version?

Anyway if we can't change this then how we suppose update for example zfcUser backwards compatible way?